gha-webhook-listener.py

```diff
@@ -371,11 +379,23 @@ def tidy_extract_directory(target_dir, cleanup_dir, versions_to_keep):
         ),
     )
 
+    parser.add_argument(
+        "--hook-script",
+        type=str,
+        help=(
+            "Script to run after each workflow run is processed, "
+            "will be passed full path to the extracted artifact as an argument."
+        ),
+    )
+
     args = parser.parse_args()
 
     if args.keep_versions is not None and args.keep_versions < 1:
         parser.error("keep-versions should be unset or > 0")
 
+    if args.hook_script is not None and not os.path.exists(args.hook_script.split(" ", 1)[0]):
+        parser.error("given hook-script cannot be found")
```

Comment:
hrm, it's worth noting that `subprocess.run(...)` will search `$PATH`, so this will both reject some things that `subprocess.run` will accept, and vice versa.

Plus you're not checking the executable bit, checking permissions, etc etc. And even if this check passes at startup, it can still fail later.

Basically, I'm not entirely sure this check is worthwhile.

Reply:
Sure, but it helps when you fat-finger the path, its not intended to be fool-proof

Reply:
well, feel free to keep it if you prefer!

Reply:
Since it doesn't account for $PATH it isn't super useful